cohorts/load.py

```diff
@@ -306,6 +307,78 @@ def as_dataframe(self, join_with=None, join_how=None):
                 len(df)))
         return df
 
+    def as_dataframe(self, on=None, col=None, join_with=None, join_how=None, **kwargs):
+        """
+        Return this Cohort as a DataFrame, and optionally include additional columns
+        using `on`.
+
+        on : function or list or dict or str, optional
+            - A function that creates a new column for comparison, e.g. count.snv_count.
+            - Or a list of column-generating functions.
+            - Or a map of new column names to their column-generating functions.
+            - A column name that gets returned with the original dataframe as (col, df).
+        col : str, optional
+            If `on` is a function generating a column, col is the name of that column.
+            If `on` is a str specifying a column, col is the name of a copy of that column.
+            If None, defaults to the name of the function.
+
+        If `on` is a function or functions, kwargs is passed to those functions.
+        Otherwise kwargs is ignored.
+
+        Return : tuple or DataFrame
+            <dataframe>
+            or (<name of new column>, <dataframe with that new column>)
+            or (<list of names of new columns>, <dataframe with those new columns>)
+        """
+        df = self._as_dataframe_unmodified(join_with=join_with, join_how=join_how)
+        if on is None:
+            return df
+
+        if type(on) == str:
+            if col is not None:
+                df[col] = df[on]
+                return (col, df)
+            return (on, df)
+
+        def apply_func(on, col, df):
+            """
+            Sometimes we have functions that, by necessity, have more parameters
+            than just `row`. For a function like that, we assume that it has two
+            parameters: `row` and `cohort`. We use it to construct another function
+            that has only the `row` parameter so it can be sent to `DataFrame.apply`.
+            """
+            if col is None:
+                # Use the function name, or "column" for lambdas, if no
+                # name is provided for the newly created column.
+                col = on.__name__ if not is_lambda(on) else "column"
+            on_argcount = on.__code__.co_argcount
```

Comment:
Nor did I; not sure what the performance implications are, but not particularly worried for this use case.